### PR TITLE
Reduce number of `mul!` methods

### PR DESCRIPTION
**Pull request by dkarrasch**
Cf. https://github.com/JuliaGPU/GPUArrays.jl/pull/472 and https://github.com/JuliaGPU/CUDA.jl/pull/1904. Same here: if `gemm_dispatch!` is (sufficiently) internal, it might be actually removed.

### CODE REVIEW

**Comment by dkarrasch** [comment]
This requires the GPUArrays.jl PR to be merged and released first.

**Comment by maleadt** [comment]
> This requires the GPUArrays.jl PR to be merged and released first.

We commit a Manifest here, so you can actually point it to your GPUArrays branch (or GPUArrays#master, once that PR would be merged).

**Comment by dkarrasch** [comment]
I couldn't find a `generic_matvecmul!`-like function that dispatches to `gemv!` and the symmetric/hermitian counterparts. Should there be one? It seems that currently `mul!(::oneVector, ::oneMatrix, ::oneVector)` falls back to `GPUArrays.jl`. Only if the matrix is `Hermitian` calls are caught and sent to `symv!` or `hemv!`.

File: lib/mkl/linalg.jl [hunk 1]
```diff
@@ -71,6 +93,7 @@ end
 # BLAS 2
 #
 
+# TODO: Should there be a LinearAlgebra._generic_matvecmul! that dispatches to gemv!, symv! and hemv! ?
```
Comment (dkarrasch) [inline lib/mkl/linalg.jl:96]: I wonder...

**Comment by maleadt** [comment]
Same difference in ArgumentError as with the CUDA.jl PR.

> I couldn't find a generic_matvecmul!-like function that dispatches to gemv! and the symmetric/hermitian counterparts. Should there be one? It seems that currently mul!(::oneVector, ::oneMatrix, ::oneVector) falls back to GPUArrays.jl.

Yeah that looks like an oversight. oneMKL has `gemv`, so we should wrap that: https://spec.oneapi.io/versions/latest/elements/oneMKL/source/domains/blas/gemv.html

**Pull request merged**